mindsdb/integrations/handlers/spacy_handler/spacy_handler.py

```diff
@@ -0,0 +1,122 @@
+import spacy
+import pandas as pd
+from mindsdb.integrations.libs.base import BaseMLEngine
+from typing import Optional
+
+
+class SpacyHandler(BaseMLEngine):
+    """
+    Integration with the spaCy NLP library.
+    """
+
+    name = 'spacy'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.default_linguistic_feature = 'ner'
+        self.linguistic_features = ['ner', 'lemmatization', 'dependency-parsing', 'pos-tag', 'morphology']
+
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+        if 'using' not in args:
+            raise Exception("spaCy engine requires a USING clause! Refer to its documentation for more details.")
+        else:
+            args = args['using']
+
+        if len(set(args.keys()) & {'linguistic_feature', 'target_column'}) == 0:
+            raise Exception('`linguistic_feature` and `target_column` are required for this engine.')
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[dict] = None) -> None:
+        model_args = args['using']
+        model_args['target'] = target
+
+        if not args.get('linguistic_feature'):
+            args['linguistic_feature'] = self.default_linguistic_feature
+
+        train_model = args.get('train_model', False)
+        if train_model:
+            pass
```

Comment:
Is this correct? Would the model be trained at inference time then?

Reply:
Oh yeah, then it is not correct. I will delete those lines of code. With training the model, that would be in the update() function right. 

One of the use cases for training the model is for example when analysing articles regarding science. We could train NER model to adapt to recognise jargons in more scientific articles.

Reply:
> With training the model, that would be in the update() function right.
Seems not. My question was along the lines of whether you needed to do any "normal" ML training on the lang model that is downloaded by SpaCy, but it looks like it is not needed. Which means that simply removing the lines above as you did is enough here.

Reply:
And yes, for future versions we could add the option of doing `finetune()` if SpaCy allows for it. But definitely out of scope for this PR I think.